infra/Pos/Diffusion/Subscription/Common.hs

```diff
@@ -94,6 +116,19 @@ subscribeTo keepAliveTimer sendActions peer = do
     msgSubscriptionTerminated :: NodeId -> SubscriptionTerminationReason -> Text
     msgSubscriptionTerminated = sformat $ "subscriptionWorker: lost connection to "%shown%" "%shown
 
+    timeSpecToMilliseconds :: TimeSpec -> Millisecond
+    timeSpecToMilliseconds = fromMicroseconds . div 1000 . toNanoSecs
+
+    alterPeerSubStatus :: Maybe SubscriptionStatus -> m ()
+    alterPeerSubStatus s = atomically $ do
+        stats <- readTVar subStatus
+        let !stats' = Map.alter fn peer stats
+        writeTVar subStatus stats'
+        where
+            fn :: Maybe SubscriptionStatus -> Maybe SubscriptionStatus
+            fn Nothing = s
+            fn (Just t) = (t <>) <$> s
```

Comment:
I suppose it is possible that two subscription threads try to subscribe to the same `NodeId` (their DNS names resolve to intersecting sets) but it's a corner case and doesn't/shouldn't happen in practice. I think it's safe to just replace the value with the new one / delete if it's `Nothing`.

Reply:
I agree, this shouldn't be a problem.

Reply:
Indeed, they are isomorphic.